tools/clang/test/CodeGenSPIRV/vk.buffer-pointer.read.hlsl

```diff
@@ -0,0 +1,48 @@
+// RUN: %dxc -spirv -T ps_6_0 -E MainPs %s | FileCheck %s
+
+// CHECK: OpEntryPoint Fragment [[FUN:%[_0-9A-Za-z]*]] "MainPs" [[OUT:%[_0-9A-Za-z]*]]
+
+struct Globals_s
+{
+      float4 g_vSomeConstantA;
+      float4 g_vTestFloat4;
+      float4 g_vSomeConstantB;
+};
+
+typedef vk::BufferPointer<Globals_s> Globals_p;
```

Comment:
Looks like doing this crashes the compiler:
```
typedef vk::BufferPointer<Globals_p> Globals_pp;;

PushConstant_t {
    Globals_pp bda;
}

bda.Get().Get().g_testFloat4

```

Reply:
I fixed some code that was mishandling pointer-to-pointer situations.  Does your full example work for you now?

Reply:
Yes, still can reproduce with this snippet (linux, debug build)

```
// RUN: %dxc -E main -T cs_6_7  %s -spirv -E main | FileCheck %s

struct Globals_s {
  float4 a;
};

typedef vk::BufferPointer<Globals_s> Globals_p;
typedef vk::BufferPointer<Globals_p> Globals_pp;

[[vk::push_constant]]
Globals_pp bda;

[numthreads(1, 1, 1)]
void main() {
  float4 r = bda.Get().Get().a;
}
```

```
SpirvEmitter.cpp:10865
bufferPointer->getResutlType() -> StructType
dyn_cast<SpirvPointerType>(bufferPointer->getResultType()); yields nullptr
```